functions/packages.bash

```diff
@@ -141,477 +201,486 @@ homegear_setup() {
-        sed -i "s|KEYPATH|${keypath}|g" /etc/nginx/sites-enabled/openhab
-        uncomment "#REDIR" /etc/nginx/sites-enabled/openhab
-        comment "listen" /etc/nginx/sites-enabled/openhab
-        uncomment "#SSL" /etc/nginx/sites-enabled/openhab
-      fi
-    fi
-
-    nginx -t && systemctl restart nginx.service || FAILED=true
-    if [ "$FAILED" = true ]; then
-      whiptail --title "Operation Failed!" --msgbox "$failtext" 15 80
+      if cond_redirect apt-get install --yes certbot python3-certbot-nginx; then echo "OK"; else echo "FAILED"; return 1; fi
+
+      echo -n "$(timestamp) [openHABian] Configuring certbot... "
+      mkdir -p /var/www/"$domain"
+      uncomment "#WEBROOT" /etc/nginx/sites-enabled/openhab
+      if ! nginx -t; then echo "FAILED (Nginx configuration test)"; fi
+      if ! cond_redirect systemctl reload nginx; then echo "FAILED (Nginx reload)"; fi
+      if cond_redirect certbot certonly --webroot -w /var/www/"$domain" -d "$domain"; then echo "OK"; else echo "FAILED"; return 1; fi
+      certpath="/etc/letsencrypt/live/${domain}/fullchain.pem"
+      keypath="/etc/letsencrypt/live/${domain}/privkey.pem"
     else
-      whiptail --title "Operation Successful!" --msgbox "Setup successful. Please try entering $protocol://$domain in a browser to test your settings." 15 80
+      echo -n "$(timestamp) [openHABian] Configuring openSSL... "
+      mkdir -p /etc/ssl/certs
+      certpath="/etc/ssl/certs/openhab.crt"
+      keypath="/etc/ssl/certs/openhab.key"
+      whiptail --title "openSSL Key Generation" --msgbox "openSSL is about to ask for information in the command line, please fill out each line." 8 80 3>&1 1>&2 2>&3
+      if ! openssl req -x509 -nodes -days 3650 -newkey rsa:2048 -keyout $keypath -out $certpath; then echo "FAILED (openSSL configuration)"; fi
     fi
-  else
-    whiptail --title "Operation Canceled!" --msgbox "Setup was canceled, no changes were made." 15 80
+    uncomment "#CERT" /etc/nginx/sites-enabled/openhab
+    if ! sed -i "s|CERTPATH|${certpath}|g" /etc/nginx/sites-enabled/openhab; then echo "FAILED (certpath config)"; fi
+    if ! sed -i "s|KEYPATH|${keypath}|g" /etc/nginx/sites-enabled/openhab; then echo "FAILED (keypath config)"; fi
+    uncomment "#REDIR" /etc/nginx/sites-enabled/openhab
+    comment "listen" /etc/nginx/sites-enabled/openhab
+    uncomment "#SSL" /etc/nginx/sites-enabled/openhab
   fi
+
+  if ! nginx -t; then echo "FAILED (Nginx configuration test)"; fi
+  if ! cond_redirect systemctl restart nginx; then echo "FAILED (Nginx restart)"; fi
+
+  whiptail --title "Operation Successful!" --msgbox "Setup successful. Please try entering $protocol://$domain in a browser to test your settings." 8 80
 }
 
 ## Function for installing Telldus Core service for Tellstick USB devices.
 ## The function can be invoked either INTERACTIVE with userinterface or UNATTENDED.
 ##
-##    tellstick_core_setup()
+##    telldus_core_setup()
```

Comment:
did you change anything about the install process ?
can you test ? There's #633 still open.

Reply:
I can't test to make sure that the service works but the install completes properly.

Reply:
ok so we can close #633
(if it doesn't work Elias or someone to hit this can reopen)

Reply:
Yes.